src/plugins/data_explorer/public/utils/state_management/preload.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { PreloadedState } from '@reduxjs/toolkit';
+import { getPreloadedState as getPreloadedMetadataState } from './metadata_slice';
+import { RootState } from './store';
+import { DataExplorerServices } from '../../types';
+
+export const getPreloadedState = async (
+  services: DataExplorerServices
+): Promise<PreloadedState<RootState>> => {
+  let rootState: RootState = {
+    metadata: await getPreloadedMetadataState(services),
+  };
+
+  // initialize the default state for each view
+  const views = services.viewRegistry.all();
+  const promises = views.map(async (view) => {
+    if (!view.ui) {
+      return;
+    }
+
+    const { defaults } = view.ui;
+
+    // defaults can be a function or an object
+    const preloadedState = typeof defaults === 'function' ? await defaults() : defaults;
```

Comment:
I am assuming `defaults()` is external to this plugin; if so, is it work wrapping in a `try ... catch`?